### ISSUE TITLE
read timeout for ec2 discovery plugin

### ISSUE THREAD

**Issue by dat-vikash** [elastic/elasticsearch#19078]
**Elasticsearch version**: 2.3.3

**JVM version**: 1.8

**OS version**: Linux ip-10-0-1-106 4.1.13-19.31.amzn1.x86_64 #1 SMP Wed Jan 20 00:25:47 UTC 2016 x86_64 x86_64 x86_64 GNU/Linu

**Description of the problem including expected versus actual behavior**:
It seems with a large number of ec2 instances within a region, the ec2 discovery plugin will timeout

``` bash
[2016-06-25 03:48:09,987][INFO ][discovery.ec2            ] [Zartra] Exception while retrieving instance list from AWS API: Unable to unmarshall response (ParseError at [row,col]:[174,26]
Message: Read timed out). Response Code: 200, Response Text: OK
[2016-06-25 03:49:21,390][INFO ][discovery.ec2            ] [Zartra] Exception while retrieving instance list from AWS API: Unable to unmarshall response (java.net.SocketTimeoutException: Read timed out). Response Code: 200, Response Text: OK
```

It would be great if we can configure this timeout either through environmental variables of system properties.

**Comment by liketic** [comment]
@dadoonet Maybe this can be closed? 👍

**Comment by DaveCTurner** [comment]
Closed by #21956.
